### PR TITLE
update user model

### PR DESCRIPTION
**Pull request by MariyaSin**
Added City, Gender, DateOfBirth properties
Changed property StartWorkingAt to nullabe
changed positionId to required
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/704
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/702
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/701

### CODE REVIEW

**Review by borovHokage — APPROVED** [review]

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 1]
```diff
@@ -235,19 +239,18 @@ public OperationResultResponse<Guid> Execute(CreateUserRequest request)
 
             var dbUser = _mapperUser.Map(request, avatarImageId);
 
-            Guid userId = _userRepository.Create(dbUser, request.Password);
+            var password = request.Password != null ? request.Password : _generatePassword.Execute();
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:242]: You  can do it like var password = request.Password ?? _generatePassword.Execute();

File: src/UserService.Models.Db/DbUser.cs [hunk 2]
```diff
@@ -60,6 +63,10 @@ public void Configure(EntityTypeBuilder<DbUser> builder)
                 .Property(p => p.LastName)
                 .IsRequired();
 
+            builder
```
Comment (BelehovEgor) [inline src/UserService.Models.Db/DbUser.cs:66]: int can't be null, it is not requered to write this

**Review by BelehovEgor — APPROVED** [review]

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs [hunk 3]
```diff
@@ -28,12 +28,27 @@ public class PatchDbUserMapper : IPatchDbUserMapper
...
+                {
+                    result.Operations.Add(new Operation<DbUser>(item.op, $"/{nameof(DbUser.DateOfBirth)}", item.from, DateTime.Parse(item.value.ToString())));
+                    continue;
+                }
+                if (item.path.EndsWith(nameof(EditUserRequest.StartWorkingAt), StringComparison.OrdinalIgnoreCase))
+                {
+                    result.Operations.Add(new Operation<DbUser>(item.op, $"/{nameof(DbUser.StartWorkingAt)}", item.from, DateTime.Parse(item.value.ToString())));
+                    continue;
+                }
+                    result.Operations.Add(new Operation<DbUser>(item.op, item.path, item.from, item.value));
```
Comment (djino97) [inline src/UserService.Mappers/Models/PatchDbUserMapper.cs:51]: fix format

**Review by djino97 — APPROVED** [review]

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 1 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs]: Done

File: src/UserService.Models.Db/DbUser.cs (hunk 2 above)
Comment (MariyaSin) [reply src/UserService.Models.Db/DbUser.cs]: done

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs (hunk 3 above)
Comment (MariyaSin) [reply src/UserService.Mappers/Models/PatchDbUserMapper.cs:51]: done

**Pull request merged**